go/ai-proxy/api/openai/openai.go

```diff
@@ -0,0 +1,91 @@
+package openai
+
+import (
+	"encoding/json"
+	"time"
+
+	ollamaapi "github.com/ollama/ollama/api"
+	"github.com/ollama/ollama/openai"
+	"github.com/pluralsh/polly/algorithms"
+	"github.com/samber/lo"
+)
+
+type Endpoint string
+
+const (
+	EndpointChat = "/v1/chat/completions"
+)
+
+type ChatCompletionRequest struct {
+	openai.ChatCompletionRequest `json:",inline"`
+}
+
+func (in ChatCompletionRequest) MarshalJSON() ([]byte, error) {
+	return json.Marshal(struct {
+		Model    string    `json:"model"`
+		Messages []Message `json:"messages"`
+		Stream   bool      `json:"stream,omitempty"`
+	}{
+		Model: in.Model,
+		Messages: algorithms.Map(in.Messages, func(m openai.Message) Message {
+			return Message{
+				Message: m,
+			}
+		}),
+		Stream: in.Stream,
+	})
+}
+
+type Message struct {
+	openai.Message `json:",inline"`
+}
+
+func (in Message) MarshalJSON() ([]byte, error) {
+	return json.Marshal(struct {
+		Role    string `json:"role"`
+		Content any    `json:"content"`
+	}{
+		Role:    in.Role,
+		Content: in.Content,
+	})
+}
+
+func ToChatCompletionRequest(in ollamaapi.ChatRequest) ChatCompletionRequest {
+	return ChatCompletionRequest{
+		ChatCompletionRequest: openai.ChatCompletionRequest{
+			Model: in.Model,
+			Messages: algorithms.Map(
+				in.Messages,
+				func(message ollamaapi.Message) openai.Message {
+					return openai.Message{
+						Role:    message.Role,
+						Content: message.Content,
+					}
+				}),
+			Stream: lo.FromPtr(in.Stream),
+		},
+	}
+}
+
+func FromChatCompletionResponse(in openai.ChatCompletion) ollamaapi.ChatResponse {
+	return ollamaapi.ChatResponse{
+		Model:     in.Model,
+		CreatedAt: time.Unix(in.Created, 0),
+		Message:   toFlatMessage(in.Choices),
+		Done:      true,
+	}
+}
+
+func toFlatMessage(choices []openai.Choice) ollamaapi.Message {
+	var result ollamaapi.Message
+	if len(choices) == 0 {
+		return result
+	}
+
+	choice := choices[0]
```

Comment:
yeah this is basically the same logic I'm using in the main console so should work.